### ISSUE TITLE
feat(model): integrate pinned V4.1 text decoder and sessions

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12909]
## Current state

Parent: #12640. Metadata prerequisite #12803 is merged at 2d2bca9c5dd567488909a70e807e2706c9edd01c. Native V4.1 execution remains refused before weight I/O. No decoder leaf was found in open/closed V4.1 issue search on 2026-09-12.

## Scope

Implement the V4.1 text decoder/session path in internal/model: exact text input compatibility, single-pass mHC, shared/routed MoE, norm/head/logits, and ordinary Prefill/Step dispatch. Pin equations to deepseek-ai/DeepSeek-V4.1-Flash@dba1be0a40aa45a94ad051997016db3960a90277/inference/model.py and encoding/. Retain the native unsupported guard until the complete path has independent parity evidence. V4 Flash 0731 equations are not V4.1 evidence.

Coordinates with: #12640 (sibling loader, Engram, attention/cache and independent parity leaves).
Promotion requires: independently witnessed loader, Engram, attention/cache contracts and full token-path parity. Disjoint source inspection and mHC/MoE computation can proceed before integration.

## User outcome

Priority tier: 2, serving. Centrality: Core.
For: native local text inference users.
Problem: metadata admission cannot execute V4.1 tokens.
Today: fak refuses the native model; the pinned official reference supplies the behavioral specification.
Better because: a native decoder retains fak ownership of model execution and state.

## Likely files

internal/model/v41_decoder.go, internal/model/v41_decoder_test.go; minimal coordinated hooks in the existing model/session dispatch. Dedicated V4.1 fixtures only. Loader, Engram and attention implementations belong to sibling leaves.

## Definition of done

The ordinary loader/session path produces independently derived tiny-fixture logits through Prefill and Step, including mHC and routed/shared MoE. Malformed inputs fail closed, and relevant V4 regressions remain green. Until every component is witnessed, this issue remains open and native loading remains refused. Fixture evidence makes no physical checkpoint qualification claim.

## Witness

`./test.ps1 ./internal/model -run 'TestDeepSeekV41|TestV41|TestV4' -count=1`

Also run affected-package vet and required boundary/leak checks before landing.

## Work estimate

8 points. Completion standard: integrated.

## Overall completion contribution

8/40 points
Parent context: #12640

## Target operating envelope

- complete fixture token paths: >= 1 paths

## Witnessed operating envelope

- complete fixture token paths: = 0 paths

## Core through-line

Wire the pinned V4.1 text token path into ordinary native sessions with independently verified logits.

## Gold-plating boundary

No vision, DSpark, physical qualification, full checkpoint downloads, or performance tuning. Preserve the unsupported loader guard while sibling integration is incomplete.

## Problem frame

Centrality: Core
P1: advanced - Preserve sequential decoder and shared-cache state through Prefill/Step; compare incremental and prefill logits.
P2: preserved - Require numeric parity before efficiency claims; tiny fixtures are software witnesses only.
P3: advanced - Add family-specific bounded seams without changing V4 Flash semantics; reject malformed geometry.
P4: advanced - Integrate the ordinary native loader/session path and preserve fail-closed admission until supported.

## Parent context

#12640 native DeepSeek V4.1 Flash execution; estimated 40-point text integration baseline for this scope.

## Scope class
S1

- [ ] Implement pinned mHC, routed/shared MoE and norm/head with independent numeric and malformed-input fixtures.
- [ ] Close the direct Model/Session admission bypass before generic math or weight access.
- [ ] Integrate loader, Engram and attention on the ordinary Prefill/Step path and prove tiny complete-path parity.
- [ ] Prove official tokenizer/encoding compatibility separately from raw token IDs.

## Completion standard
integrated

**Comment by anthony-chaudhary** [comment]
Progress receipt (2026-09-12), source dba1be0a40aa45a94ad051997016db3960a90277:

- Direct Model/Session native admission bypass reproduced with TestDeepSeekV41DirectAdmission, then repaired in kv.go/forward.go/hal.go. Focused WSL V4.1 and V4 regressions passed. Isolated validation passed build/vet/test; overall validation remains refused at Strix controller authority (UNSUPPORTED_VALIDATOR_BINARY), with canonical stamped WSL recovery coordinated across siblings.
- Scalar mHC, shared/routed MoE, and norm/head helpers implemented behind a row-reader contract. Independent source review found and is correcting repeated reader-dimension queries; numeric parity is checked by #12908, with no full-token or checkpoint claim.
- Decoder owns official tokenizer normalization/map acceptance. The exact source builder and pinned tokenizer, using tokenizers 0.22.2, produce 129280 entries and 99092 compressed IDs. Derived LE uint32 map SHA256 c60a86322ec17b4142bfef3c57a8d81fb428550cdf487f88a4320cb59fe46481; tokenizer SHA256 c90dfa01249db1be4245780a052ede752e1361c612ac6d08e2bdada7d599476b. Native V41OfficialTokenMap validates exact tokenizer fingerprint, embedded-map digest, dimensions, canonical ordering, and fresh output ownership. Its public API passes with the actual pinned tokenizer. Independent separate-algorithm map reconstruction remains pending at #12908.
- Existing tokenizer misroutes the pinned three-stage V4.1 Split pipeline to GLM4. A source-derived split regression is red for CJK/combining/symbol boundaries; exact V4.1 dispatch is being implemented.

Coordinates with: #12892 (loader), #12893 (Engram), #12890 (attention), #12908 (independent parity).
Promotion requires: #12892, #12893, #12890, #12908 plus completed V4.1 chat serialization and ordinary Prefill/Step wiring.

The native unsupported guard remains closed. Full checkpoint FP4 row execution, full chat encoding, Engram projection/gate, and complete session integration remain unfulfilled. This issue and #12640 remain open. No PR, full checkpoint download, or physical qualification claim.

**Comment by anthony-chaudhary** [comment]
Implementation remains in progress. The text chat encoder now passes independent golden cases from the pinned official encoder. The Engram projection/gate and scalar session scheduling have targeted software tests. Native execution remains guarded: packed FP4 execution, attention integration, ordinary session wiring, and complete-path verification are still required. This issue remains open; no full-checkpoint or physical qualification is claimed.

**Comment by anthony-chaudhary** [comment]
Integration prerequisite recorded from the #12937 review: `V41EngramShard.ID` and `ExpectedID` are intentionally metadata labels; the low-level `ReaderAt` constructor does not authenticate the backing bytes. This is not a #12937 blocker under its documented preverified-caller contract, and hashing the roughly 203 GB table during each constructor call would defeat bounded streaming.

Before #12909 removes native admission refusal, the loader must bind each opened reader/range to the pinned checkpoint-index shard name, exact file size, tensor offset/length, and an immutable artifact digest/provenance receipt established during acquisition or explicit artifact verification. Pass that observed identity separately from the index-requested identity. Add a loader-bound negative fixture proving a wrong shard/range cannot reach Engram projection or `Prefill`/`Step`. Tiny fixtures may use a byte-hashed miniature shard. This is an integration prerequisite, not end-to-end readiness evidence.

**Comment by anthony-chaudhary** [comment]
Landed: the V4.1 text decoder output tail and ordinary Prefill/Step seam.

- Commit: eab12fc72 (\eat(ticket): complete flow ticket 12909 (12909)\) on \main\ (pushed).
- Files: \internal/model/v41_decoder.go\, \internal/model/v41_decoder_test.go\ (709 insertions).
- Delivers the pinned final RMSNorm + output-head projection + logits for the 40-layer/width-5120 geometry, driven by \V41TextDecoder.Prefill([]int)/Step(id)\ over a persistent \V41DecodeState\; malformed geometry/token ids fail closed with a typed error wrapping \ErrV41DecoderGeometry\/\ErrV41DecoderToken\. V4 Flash 0731 equations are not consulted.
- Witness (green on the landed commit): \go -C ..\\fak test ./internal/model -run 'TestV41Decoder' -count=1\; full \go test ./internal/model/... -count=1\ green (202s), \go vet\ clean, \gofmt\ clean, \internal/abi\ byte-unchanged.
- Provenance: Impl/Test-Context SEPARATED (recorded on the commit).

Ref #12640
